### ISSUE TITLE
[BUG] Anchors in API reference docs are broken

### ISSUE THREAD

**Issue by septatrix** [minio/docs#780]
**Describe the bug**
All anchors in the API references pointing to methods/symbols seem to be broken.
They do not jump to the method and instead opens the reference in a new tab (which also does not jump to the method but instead stays at the top of the reference)

**To Reproduce**
1. Go to <https://min.io/docs/minio/linux/developers/javascript/API.html>
2. Click on any of the methods in the table at the start, e.g. `makeBucket`
3. a) A new tab opens, b) neither the new tab nor the old tab jumped to the method documentation

**Expected behavior**
A new tab should not be opened and the current tab should scroll to the linked method.

**Desktop (please complete the following information):**
 - OS: Linux
 - Browser: Chrome
 - Version: 111

**Additional context**
This seems to be due to some script adding `target="_blank"` to every link. With JS disabled the whole page works mostly fine and the link properly work. I am not sure why a script is adding the target to every link though I would guess that it is some erroneous regex going rouge which is supposed to only links to external websites.

**Comment by ravindk89** [comment]
So I tested this in Firefox 110 - I get a new tab, but it does scroll properly. 

That said we might need to do a bit of checking here - these are imported markdown pages, and we're relying on MyST to do some of the heavy lifting for us.

@rushenn any ideas on your end? I'll poke around here too.

**Comment by rushenn** [comment]
<img width="742" alt="Screenshot 2023-04-01 at 14 38 32" src="https://user-images.githubusercontent.com/13393018/229282160-f9312df2-d194-4f40-8ee0-a615d810dadf.png">

It seems like Sphinx is incorrectly identifying these links as external. I believe this to be a bug in Sphinx, I will send a PR to correct this @ravindk89.

**Comment by ravindk89** [comment]
It might also be the MyST parser since that is where those markdown files
get transformed.

If you have an idea of where that code path executes then by all means.
Otherwise I can reach out to the maintainer in question or try updating myst

**Comment by rushenn** [comment]
@ravindk89 Correct, it might be the parser. I was thinking to give a JS treatment but It would be great if you can reach out the maintainers and try updating the parser itself.

**Comment by ravindk89** [comment]
I'll take a poke, this could also very well be a config issue on my end.
Will see.

**Comment by ravindk89** [comment]
Not at this time. We have limited cycles available to debug this, and we also need to not lose the linking functionality entirely (as for a while that was *also* completely broken). 

We will update when we can. We appreciate your patience until then.

**Comment by ravindk89** [comment]
Notes: Updating to myst-parser 1.0.0 didn't seem to do much. Also set `myst_all_links_external` to `False` and these links still get generated with `external`, which is very odd. 

```
myst v1.0.0: MdParserConfig(commonmark_only=False, gfm_only=True, enable_extensions=set(), disable_syntax=[], all_links_external=False, url_schemes=('http', 'https', 'mailto', 'ftp'), ref_domains=None, fence_as_directive=set(), number_code_blocks=[], title_to_header=False, heading_anchors=2, heading_slug_func=None, html_meta={}, footnote_transition=True, words_per_minute=200, substitutions={}, linkify_fuzzy_links=True, dmath_allow_labels=True, dmath_allow_space=True, dmath_allow_digits=True, dmath_double_inline=False, update_mathjax=True, mathjax_classes='tex2jax_process|mathjax_process|math|output_area', enable_checkboxes=False, suppress_warnings=[], highlight_code_blocks=True)

```

I don't see anything in here that would cause this behavior coming from Myst.

**Comment by ravindk89** [comment]
Updating to Sphinx 6.3.2 also didn't help.

@rushenn I'm not sure if javascript can help us here - there's nothing on this page that would distinguish the developer Markdown-based quickstarts from anything else. If we add JS to strip out the external tag, it would likely apply to all pages. Any ideas?

**Comment by ravindk89** [comment]
OK so the issue seems to be that setting `myst_gfm_only` converts all links into externals. We also get a few hundred warnings about `myst.xref_missing`. 

The links *seem* to work however. If I can figure out how to suppress the warnings I think we are good.

**Comment by ravindk89** [comment]
Warnings suppressed 

@feorlen @djwfyi this required upgrading to nearly-latest stable Sphinx. Build times are much better now though, and some of our warning outputs look more useful.

Please test this branch thoroughly and see what else I might have missed. PR to open shortly.

**Comment by septatrix** [comment]
Thanks for taking a look at it.

> OK so the issue seems to be that setting myst_gfm_only converts all links into externals. We also get a few hundred warnings about myst.xref_missing.

Seems to be due to the following lines: https://github.com/executablebooks/MyST-Parser/blob/a784880c7fbd893566087b83deec081376740e2c/myst_parser/mdit_to_docutils/base.py#L907-L912. It might make sense to report this upstream as that is not how GFM works to my knowledge. E.g. I can link [to the top of the discussion](#partial-discussion-header) easily.

> The links seem to work however.

Could you also please test on chrome based browsers (or will I be able to test once the PR is open)? You said the status quo works for you in firefox (it also does for me) though chrome is broken so maybe chrome handles following anchors in new tabs differently. That would also still leave the issue of docs being opened in a new tab.

**Comment by djwfyi** [comment]
@septatrix My testing was in Chrome. The links in the API docs worked fine.

**Comment by ravindk89** [comment]
That's a great catch @septatrix - I'll report that upstream.

**Comment by ravindk89** [comment]
So for now this is fixed with #844 , so closing this out.
